modeling/theoretical-simulator.go

```diff
@@ -27,48 +30,142 @@ func (sim TheoreticalSimulator) Simulate(region regionInfo, endpointSlices map[s
 	if len(region.zoneDetails) == 0 || len(endpointSlices) == 0 {
 		return SimulationResult{}, errors.New("can't evaluate probability based on empty zones or endpointslices")
 	}
-	// kube-proxy calculation, zones name - endpoints in potential destination sclies
-	kubeProxy := make(map[string]map[string]float64)
-	for name := range region.zoneDetails {
-		kubeProxy[name] = map[string]float64{}
-		for _, slice := range endpointSlices {
-			for compZone, pods := range slice.Composition {
-				kubeProxy[name][compZone] += float64(pods) * slice.Weights[name]
+	// weighted number of endpoints in each sliceGroup can be reached by every
+	// zone
+	// first key: zone name
+	// second key: sliceGroup label
+	zoneReachableEndpointsInSG := make(map[string]map[string]float64)
+	for zone := range region.zoneDetails {
+		zoneReachableEndpointsInSG[zone] = map[string]float64{}
+		for sliceLabel, slice := range endpointSlices {
+			zoneReachableEndpointsInSG[zone][sliceLabel] = float64(slice.numberOfEndpoints()) * slice.ZoneTrafficWeights[zone]
```

Comment:
What if `slice.ZoneTrafficWeights[zone]` is not set?

Reply:
if not set, deem as 0. if its a mistake, this should not be the responsibility for this function to take. The creator should take care of the weights value and set it correctly.